include/seqan3/alignment/pairwise/execution/alignment_executor_two_way.hpp

```diff
@@ -44,7 +46,8 @@ template <std::ranges::ViewableRange resource_t,
           typename alignment_algorithm_t,
           typename execution_handler_t = execution_handler_sequential>
 //!\cond
-    requires std::ranges::InputRange<resource_t> &&
+    requires ((std::ranges::InputRange<resource_t> && std::ranges::SizedRange<resource_t>) ||
+             std::ranges::ForwardRange<resource_t>) &&
```

Comment:
Are there pure input ranges that are sized? Why was it necessary to change this?

Reply:
Well I had once the idea to also allow a generating views or something like this to get the sequences. but for the parallel interface I do need the Size information before. Otherwise I would consume an InputRange by calling std::ranges::distance. This would ensure that it works as well. But I also agree that we are talking about very rare use cases and from that point it would be simply safer to require ForwardRange. I will change this.